wordpress-mobile/WordPress-iOS
# Decoupling Authentication Flow: Mark I

### Details:
In this PR we're taking a first step towards encapsulating all of the WordPress Authentication-Y code into it's own framework:

- Renamed **SigninHelpers** >> **WordPressAuthenticator**.
- Implemented a mechanism to relay back to the host app all of the internal events being tracked by means of Tracks. **WordPressAuthenticator** will **not** talk to Tracks, directly.
- Implemented **WordPressAuthenticationTracker**, which will (always) live in the WPiOS side of things. This class is a simple wrapper, which will be in charge of listening Flow-Y events, and relay those events back to Tracks.

### To test:
1. Verify that the unit tests are still happy
2. Run a smoke test over the login flow, verify nothing breaks violently

Needs Review: @nheagy 
cc @astralbodies 

Sir, may i bug you with a hopefully fun review?
Thanks in advance!
